pkg/controller.v1alpha3/suggestion/composer/composer.go

```diff
@@ -109,6 +116,36 @@ func (g *General) desiredContainer(s *suggestionsv1alpha3.Suggestion) (*corev1.C
 		Name: consts.ContainerSuggestion,
 	}
 	c.Image = suggestionContainerImage
+	c.Ports = []corev1.ContainerPort{
+		{
+			Name:          consts.DefaultSuggestionPortName,
+			ContainerPort: consts.DefaultSuggestionPort,
+		},
+	}
+	c.ReadinessProbe = &corev1.Probe{
+		Handler: corev1.Handler{
+			Exec: &corev1.ExecAction{
+				Command: []string{
+					defaultGRPCHealthCheckProbe,
+					fmt.Sprintf("-addr=:%d", consts.DefaultSuggestionPort),
+					fmt.Sprintf("-service=%s", consts.DefaultGRPCService),
+				},
+			},
+		},
+		InitialDelaySeconds: defaultInitialDelaySeconds,
+	}
+	c.LivenessProbe = &corev1.Probe{
```

Comment:
Will liveness probe has any hidden consequences?  Say if service is not accessible for some time

Reply:
Yeah, it will. But I think we should have it. Since we can mark the deployment unavailable then avoid timeout requests

Reply:
sound good